internal/test/wmcb_test.go

```diff
@@ -0,0 +1,224 @@
+	// remote powershell session opened
+	remotePowerShellCmdPrefix = "powershell.exe -NonInteractive -ExecutionPolicy Bypass "
+	// winrm port to be used
+	winRMPort = 5986
+)
+
+// testFramework holds the info to run the test suite
+type testFramework struct {
+	// credentials to access the Windows VM created
+	credentials *types.Credentials
+	// winrmClient to access the Windows VM created
+	winrmClient *winrm.Client
+	// fileNames contains a list of files to be transferred to the Windows VM
+	fileNames []string
+	// remoteDir is the directory to which files will be transferred to, on the Windows VM
+	remoteDir string
+	// sshClient contains the ssh client information to access the Windows VM via ssh
+	sshClient *ssh.Client
+	// cloudProvider holds the information related to cloud provider
+	cloudProvider cloudprovider.Cloud
+}
+
+// framework holds the instantiation of test suite being executed. As of now, temp dir is hardcoded.
+// TODO: Create a temporary remote directory on the Windows node
+var framework = &testFramework{remoteDir: "C:\\Temp"}
+
+// binaryToBeTransferred holds the binary that needs to be transferred to the Windows VM
+// TODO: Make this an array later with a comma separated values for more binaries to be transferred
+var binaryToBeTransferred = flag.String("binaryToBeTransferred", "",
+	"Absolute path of the binary to be transferred")
+
+// setup sets up the initial test cluster for the Windows
+// TODO: move this to return error and catch assertions there
+func (f *testFramework) setup() {
+	if err := framework.createWindowsVM(); err != nil {
+		log.Fatalf("failed spinning up Windows VM with error: %v", err)
+	}
+	// TODO: Add some options to skip certain parts of the test
+	if err := framework.setupWinRMClient(); err != nil {
+		log.Fatalf("failed setting up winRM client for the Windows VM with error: %v", err)
+	}
+	// Wait for some time before starting configuring of ssh server. This is to let sshd service be available
+	// in the list of services
+	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
+	// has those services present
+	time.Sleep(time.Minute)
+	if err := framework.configureOpenSSHServer(); err != nil {
+		log.Fatalf("failed configuring OpenSSHServer on the Windows VM created with error: %v", err)
+	}
+	if err := framework.createRemoteDir(); err != nil {
+		log.Fatalf("failed to create remote dir with error: %v", err)
+	}
+	if err := framework.getSSHClient(); err != nil {
+		log.Fatalf("failed getting ssh client for the Windows VM created with error: %v", err)
+	}
+}
+
+// createWindowsVM spins up the Windows VM in the given cloud provider and gives us the credentials to access the
+// windows VM created
+func (f *testFramework) createWindowsVM() error {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	awsCredentials := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	dir := os.Getenv("ARTIFACT_DIR")
+	privateKeyPath := os.Getenv("KUBE_SSH_KEY_PATH")
```

Comment:
What will happen if this is run where these environment variables are not present? Shouldn't we fail gracefully?